### DISCUSSION TITLE
Can not update connected devices

### DISCUSSION THREAD

**Discussion by serrov** [vmakeev/huawei_mesh_router#71] [Q&A] [answered]
Роутер: WS7100
Не работает: Отслеживание устройств

Ошибка:
Logger: custom_components.huawei_mesh_router.update_coordinator (Huawei Router)
Source: custom_components/huawei_mesh_router/update_coordinator.py:203

Can not update connected devices: UnicodeDecodeError('utf-8', b'[{"ID":"InternetGatewayDevice.LANDevice.1.Hosts.Host.2.","MACAddress":"D0:37:45:AF.....

**Comment by vmakeev** [comment]
Странная история, словно в ответе от роутера данные приходят в какой-то другой кодировке. Сначала грешил на возможную кириллицу в именах девайсов, но воспроизвести поведение не сумел.

Мне потребуется больше логов для анализа ситуации. Собрать их можно следующим образом:

1. Включить отладочное логирование
![image](https://github.com/vmakeev/huawei_mesh_router/assets/13693034/bed6f47c-cbd3-4e1d-b0b6-1b385bbc739d)

2. Подождать некоторое время пока в логах снова не появится эта ошибка

3. Выключить отладочное логирование
![image](https://github.com/vmakeev/huawei_mesh_router/assets/13693034/4740c156-6fff-4e7d-8ad0-c77187afabb2)

Полученный файл необходимо отправить мне на почту v.makeev88@yandex.ru (ну или прикрепить к собщению в этом обсуждении)

Логины-пароли в логи компонента не пишутся. При желании перед отправкой можно проверить файл на их отсутствие

**Comment by serrov** [comment]
[home-assistant_huawei_mesh_router_2024-01-25T12-53-48.150Z.zip](https://github.com/vmakeev/huawei_mesh_router/files/14052355/home-assistant_huawei_mesh_router_2024-01-25T12-53-48.150Z.zip)

**Comment by vmakeev** [comment]
Кажется я понял в чем проблема. Некоторые устройства имеют имена типа `User-\xe5\xbc\xb7`, которые штатная библиотека не может преобразовать в корректный текст.

Попробую исполнить пару трюков, отпишусь тут по результатам

**Comment by vmakeev** [comment, marked as answer]
@serrov кажется удалось решить эту проблему. Если не сложно, попробуйте установить предрелизную версию [v0.9.1.1-rc1](https://github.com/vmakeev/huawei_mesh_router/releases/tag/v0.9.1.1-rc1), и сообщите, помогли ли внесенные исправления компоненту функцонировать корректно

---

Explanation for English speakers: if there were specific characters in the names of devices connected to the mesh network that were not supported by the UTF-8 encoding, device tracking in the component may not have worked. This issue has been resolved in version [v0.9.1.1](https://github.com/vmakeev/huawei_mesh_router/releases/tag/v0.9.1.1).

**Comment by serrov** [comment]
Работает!
Ошибок нет, спасибо!
